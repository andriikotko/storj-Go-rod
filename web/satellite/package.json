{
  "name": "storj-satellite",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "serve": "vue-cli-service serve",
    "lint": "vue-cli-service lint && stylelint '**/*.{vue,scss}' --fix",
    "test": "vue-cli-service test:unit",
    "build": "vue-cli-service build",
    "dev": "vue-cli-service build --mode development"
  },
  "dependencies": {
    "apollo-cache-inmemory": "1.6.6",
    "apollo-client": "2.6.10",
    "apollo-link": "1.2.14",
    "apollo-link-context": "1.0.20",
    "apollo-link-error": "1.1.13",
    "apollo-link-http": "1.5.17",
    "graphql": "15.3.0",
    "graphql-tag": "2.11.0",
    "load-script": "1.0.0",
    "stripe": "8.96.0",
    "vue": "2.6.12",
    "vue-class-component": "7.2.5",
    "vue-clipboard2": "0.3.1",
    "vue-property-decorator": "9.0.0",
    "vue-router": "3.4.3",
    "vuex": "3.5.1"
  },
  "devDependencies": {
    "@babel/core": "7.11.6",
    "@babel/plugin-proposal-object-rest-spread": "7.11.0",
    "@types/segment-analytics": "0.0.32",
    "@types/node": "13.11.1",
    "@vue/cli-plugin-babel": "4.5.6",
    "@vue/cli-plugin-typescript": "4.5.6",
    "@vue/cli-plugin-unit-jest": "4.5.6",
    "@vue/cli-service": "4.5.6",
    "@vue/test-utils": "1.1.0",
    "babel-core": "7.0.0-bridge.0",
    "compression-webpack-plugin": "6.0.0",
    "jest-fetch-mock": "3.0.3",
    "node-sass": "4.14.1",
    "sass-loader": "10.0.2",
    "sinon": "9.0.3",
    "stylelint": "13.7.1",
    "stylelint-config-standard": "20.0.0",
    "stylelint-scss": "3.18.0",
    "stylelint-webpack-plugin": "2.1.0",
    "ts-jest": "26.3.0",
    "tslint": "6.1.3",
    "tslint-consistent-codestyle": "1.16.0",
    "tslint-loader": "3.5.4",
    "typescript": "4.0.2",
    "vue-svg-loader": "0.16.0",
    "vue-template-compiler": "2.6.12",
    "vue-tslint": "0.3.2",
    "vue-tslint-loader": "3.5.6",
    "webpack": "4.44.1"
  },
  "postcss": {
    "plugins": {
      "autoprefixer": {}
    }
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "not ie <= 8"
  ],
  "stylelint": {
    "plugins": [
      "stylelint-scss"
    ],
    "extends": "stylelint-config-standard",
    "rules": {
      "indentation": 4,
      "string-quotes": "single",
      "no-duplicate-selectors": true,
      "selector-max-attribute": 1,
      "selector-combinator-space-after": "always",
      "selector-attribute-operator-space-before": "never",
      "selector-attribute-operator-space-after": "never",
      "selector-attribute-brackets-space-inside": "never",
      "declaration-block-trailing-semicolon": "always",
      "declaration-colon-space-before": "never",
      "declaration-colon-space-after": "always",
      "number-leading-zero": "always",
      "function-url-quotes": "always",
      "font-family-name-quotes": "always-unless-keyword",
      "comment-whitespace-inside": "always",
      "comment-empty-line-before": "always",
      "rule-empty-line-before": "always-multi-line",
      "selector-pseudo-element-colon-notation": "single",
      "selector-pseudo-class-parentheses-space-inside": "never",
      "selector-max-type": 1,
      "font-family-no-missing-generic-family-keyword": true,
      "at-rule-no-unknown": true,
      "media-feature-range-operator-space-before": "always",
      "media-feature-range-operator-space-after": "always",
      "media-feature-parentheses-space-inside": "never",
      "media-feature-colon-space-before": "never",
      "media-feature-colon-space-after": "always"
    }
  },
  "jest": {
    "automock": false,
    "setupFiles": [
      "./jestSetup.ts"
    ],
    "globals": {
      "ts-jest": {
        "diagnostics": false
      }
    },
    "moduleFileExtensions": [
      "js",
      "jsx",
      "json",
      "vue",
      "ts",
      "tsx"
    ],
    "collectCoverage": true,
    "transform": {
      "^.+\\.js$": "babel-jest",
      "^.+\\.vue$": "vue-jest",
      ".+\\.(css|styl|less|sass|scss|png|jpg|ttf|woff|woff2)$": "jest-transform-stub",
      "^.+\\.tsx?$": "ts-jest",
      "^.+\\.svg$": "<rootDir>/tests/unit/mock/svgTransform.js"
    },
    "transformIgnorePatterns": [
      "/node_modules/(?!(apollo-client|apollo-link))"
    ],
    "moduleNameMapper": {
      "^@/(.*)$": "<rootDir>/src/$1"
    },
    "snapshotSerializers": [
      "jest-serializer-vue"
    ],
    "testMatch": [
      "**/tests/unit/**/*.spec.(js|jsx|ts|tsx)|**/__tests__/*.(js|jsx|ts|tsx)"
    ],
    "testURL": "http://localhost/"
  }
}
