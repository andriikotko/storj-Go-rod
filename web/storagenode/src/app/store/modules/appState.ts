// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export const APPSTATE_MUTATIONS = {
    TOGGLE_SATELLITE_SELECTION: 'TOGGLE_SATELLITE_SELECTION',
    TOGGLE_BANDWIDTH_CHART: 'TOGGLE_BANDWIDTH_CHART',
    TOGGLE_EGRESS_CHART: 'TOGGLE_EGRESS_CHART',
    TOGGLE_INGRESS_CHART: 'TOGGLE_INGRESS_CHART,',
    TOGGLE_PAYOUT_CALENDAR: 'TOGGLE_PAYOUT_CALENDAR',
    CLOSE_ADDITIONAL_CHARTS: 'CLOSE_ADDITIONAL_CHARTS',
    CLOSE_ALL_POPUPS: 'CLOSE_ALL_POPUPS',
    SET_DARK: 'SET_DARK',
    SET_NO_PAYOUT_INFO: 'SET_NO_PAYOUT_INFO',
    SET_LOADING_STATE: 'SET_LOADING_STATE',
    TOGGLE_PAYOUT_HISTORY_CALENDAR: 'TOGGLE_PAYOUT_HISTORY_CALENDAR',
};

export const APPSTATE_ACTIONS = {
    TOGGLE_SATELLITE_SELECTION: 'TOGGLE_SATELLITE_SELECTION',
    TOGGLE_BANDWIDTH_CHART: 'TOGGLE_BANDWIDTH_CHART',
    TOGGLE_EGRESS_CHART: 'TOGGLE_EGRESS_CHART',
    TOGGLE_INGRESS_CHART: 'TOGGLE_INGRESS_CHART',
    CLOSE_ADDITIONAL_CHARTS: 'CLOSE_ADDITIONAL_CHARTS',
    TOGGLE_PAYOUT_CALENDAR: 'TOGGLE_PAYOUT_CALENDAR',
    CLOSE_ALL_POPUPS: 'CLOSE_ALL_POPUPS',
    SET_DARK_MODE: 'SET_DARK_MODE',
    SET_NO_PAYOUT_DATA: 'SET_NO_PAYOUT_DATA',
    SET_LOADING: 'SET_LOADING',
    TOGGLE_PAYOUT_HISTORY_CALENDAR: 'TOGGLE_PAYOUT_HISTORY_CALENDAR',
};

const {
    TOGGLE_SATELLITE_SELECTION,
    TOGGLE_BANDWIDTH_CHART,
    TOGGLE_EGRESS_CHART,
    TOGGLE_INGRESS_CHART,
    CLOSE_ADDITIONAL_CHARTS,
    CLOSE_ALL_POPUPS,
    TOGGLE_PAYOUT_CALENDAR,
} = APPSTATE_MUTATIONS;

export const appStateModule = {
    state: {
        isSatelliteSelectionShown: false,
        isBandwidthChartShown: true,
        isEgressChartShown: false,
        isIngressChartShown: false,
        isPayoutCalendarShown: false,
        isDarkMode: false,
        isNoPayoutData: false,
        isLoading: true,
        isPayoutHistoryCalendarShown: false,
    },
    mutations: {
        [TOGGLE_SATELLITE_SELECTION](state: any): void {
            state.isSatelliteSelectionShown = !state.isSatelliteSelectionShown;
        },
        [TOGGLE_BANDWIDTH_CHART](state: any): void {
            state.isBandwidthChartShown = !state.isBandwidthChartShown;
        },
        [TOGGLE_EGRESS_CHART](state: any): void {
            state.isEgressChartShown = !state.isEgressChartShown;
        },
        [TOGGLE_INGRESS_CHART](state: any): void {
            state.isIngressChartShown = !state.isIngressChartShown;
        },
        [TOGGLE_PAYOUT_CALENDAR](state: any, value): void {
            state.isPayoutCalendarShown = value;
        },
        [CLOSE_ADDITIONAL_CHARTS](state: any): void {
            state.isBandwidthChartShown = true;
            state.isIngressChartShown = false;
            state.isEgressChartShown = false;
        },
        [APPSTATE_MUTATIONS.SET_DARK](state: any, value): void {
            state.isDarkMode = value;
        },
        [APPSTATE_MUTATIONS.SET_NO_PAYOUT_INFO](state: any, value): void {
            state.isNoPayoutData = value;
        },
        [APPSTATE_MUTATIONS.SET_LOADING_STATE](state: any, value): void {
            state.isLoading = value;
        },
        [CLOSE_ALL_POPUPS](state: any): void {
            state.isSatelliteSelectionShown = false;
        },
        [APPSTATE_MUTATIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR](state: any, value): void {
            state.isPayoutHistoryCalendarShown = value;
        },
    },
    actions: {
        [APPSTATE_ACTIONS.TOGGLE_SATELLITE_SELECTION]: function ({commit, state}: any): void {
            if (!state.isSatelliteSelectionShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_SATELLITE_SELECTION);

                return;
            }

            commit(APPSTATE_MUTATIONS.CLOSE_ALL_POPUPS);
        },
        [APPSTATE_ACTIONS.TOGGLE_PAYOUT_CALENDAR]: function ({commit, state}: any, value: boolean): void {
            commit(APPSTATE_MUTATIONS.TOGGLE_PAYOUT_CALENDAR, value);
        },
        [APPSTATE_ACTIONS.TOGGLE_EGRESS_CHART]: function ({commit, state}: any): void {
            if (!state.isBandwidthChartShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);
                commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);

                return;
            }

            commit(APPSTATE_MUTATIONS.TOGGLE_BANDWIDTH_CHART);
            commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);
        },
        [APPSTATE_ACTIONS.TOGGLE_INGRESS_CHART]: function ({commit, state}: any): void {
            if (!state.isBandwidthChartShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);
                commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);

                return;
            }

            commit(APPSTATE_MUTATIONS.TOGGLE_BANDWIDTH_CHART);
            commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);
        },
        [APPSTATE_ACTIONS.SET_DARK_MODE]: function ({commit}: any, value: boolean): void {
            commit(APPSTATE_MUTATIONS.SET_DARK, value);
        },
        [APPSTATE_ACTIONS.SET_NO_PAYOUT_DATA]: function ({commit}: any, value: boolean): void {
            commit(APPSTATE_MUTATIONS.SET_NO_PAYOUT_INFO, value);
        },
        [APPSTATE_ACTIONS.SET_LOADING]: function ({commit}: any, value: boolean): void {
            value ? commit(APPSTATE_MUTATIONS.SET_LOADING_STATE, value) :
                setTimeout(() => { commit(APPSTATE_MUTATIONS.SET_LOADING_STATE, value); }, 1000);
        },
        [APPSTATE_ACTIONS.CLOSE_ADDITIONAL_CHARTS]: function ({commit}: any): void {
            commit(APPSTATE_MUTATIONS.CLOSE_ADDITIONAL_CHARTS);
        },
        [APPSTATE_ACTIONS.CLOSE_ALL_POPUPS]: function ({commit}: any): void {
            commit(APPSTATE_MUTATIONS.CLOSE_ALL_POPUPS);
        },
        [APPSTATE_ACTIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR]: function ({commit, state}: any, value: boolean): void {
            commit(APPSTATE_MUTATIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR, value);
        },
    },
};
