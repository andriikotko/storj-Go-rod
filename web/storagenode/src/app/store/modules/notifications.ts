// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import {
    Notification,
    NotificationsApi,
    NotificationsCursor,
    NotificationsState,
} from '@/app/types/notifications';

export const NOTIFICATIONS_MUTATIONS = {
    SET_NOTIFICATIONS: 'SET_NOTIFICATIONS',
    MARK_AS_READ: 'MARK_AS_READ',
    READ_ALL: 'READ_ALL',
    SET_LATEST: 'SET_LATEST',
};

export const NOTIFICATIONS_ACTIONS = {
    GET_NOTIFICATIONS: 'getNotifications',
    MARK_AS_READ: 'markAsRead',
    READ_ALL: 'readAll',
};

/**
 * creates notifications module with all dependencies
 *
 * @param api - payments api
 */
export function makeNotificationsModule(api: NotificationsApi) {
    return {
        state: new NotificationsState(),
        mutations: {
            [NOTIFICATIONS_MUTATIONS.SET_NOTIFICATIONS](state: NotificationsState, notificationsResponse: NotificationsState): void {
                state.notifications = notificationsResponse.notifications;
                state.pageCount = notificationsResponse.pageCount;
                state.unreadCount = notificationsResponse.unreadCount;
            },
            [NOTIFICATIONS_MUTATIONS.SET_LATEST](state: NotificationsState, notificationsResponse: NotificationsState): void {
                state.latestNotifications = notificationsResponse.notifications;
            },
            [NOTIFICATIONS_MUTATIONS.MARK_AS_READ](state: NotificationsState, id: string): void {
                state.notifications = state.notifications.map((notification: Notification) => {
                    if (notification.id === id) {
                        notification.markAsRead();
                    }

                    return notification;
                });
            },
            [NOTIFICATIONS_MUTATIONS.READ_ALL](state: NotificationsState): void {
                state.notifications = state.notifications.map((notification: Notification) => {
                    notification.markAsRead();

                    return notification;
                });

                state.unreadCount = 0;
            },
        },
        actions: {
            [NOTIFICATIONS_ACTIONS.GET_NOTIFICATIONS]: async function ({commit}: any, cursor: NotificationsCursor): Promise<NotificationsState> {
                const notificationsResponse = await api.get(cursor);

                commit(NOTIFICATIONS_MUTATIONS.SET_NOTIFICATIONS, notificationsResponse);

                if (cursor.page === 1) {
                    commit(NOTIFICATIONS_MUTATIONS.SET_LATEST, notificationsResponse);
                }

                return notificationsResponse;
            },
            [NOTIFICATIONS_ACTIONS.MARK_AS_READ]: async function ({commit}: any, id: string): Promise<any> {
                await api.read(id);

                commit(NOTIFICATIONS_MUTATIONS.MARK_AS_READ, id);
            },
            [NOTIFICATIONS_ACTIONS.READ_ALL]: async function ({commit}: any): Promise<any> {
                await api.readAll();

                commit(NOTIFICATIONS_MUTATIONS.READ_ALL);
            },
        },
    };
}
