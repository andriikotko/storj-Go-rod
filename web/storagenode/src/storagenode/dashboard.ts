// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Dashboard encapsulates dashboard stale data
 */
export class Dashboard {
    public constructor(
        public nodeID: string,
        public wallet: string,
        public satellites: SatelliteInfo[],
        public diskSpace: DiskSpaceInfo,
        public bandwidth: BandwidthInfo,
        public lastPinged: Date,
        public startedAt: Date,
        public version: string,
        public allowedVersion: string,
        public isUpToDate: boolean,
    ) { }
}

/**
 * SatelliteInfo encapsulates satellite ID, URL and disqualification
 */
export class SatelliteInfo {
    public constructor(
        public id: string = '',
        public url: string = '',
        public disqualified: Date | null = null,
        public suspended: Date | null = null,
    ) { }
}

/**
 * DiskSpaceInfo stores all info about storage node disk space usage
 */
export class DiskSpaceInfo {
    public constructor(
        public used: number = 0,
        public available: number = 0,
        public trash: number = 0,
    ) {}
}

/**
 * BandwidthInfo stores all info about storage node bandwidth usage
 */
export class BandwidthInfo {
    public constructor(
        public used: number,
    ) { }
}
