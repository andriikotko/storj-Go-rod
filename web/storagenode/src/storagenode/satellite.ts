// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Satellite encapsulates satellite related data
 */
export class Satellite {
    public constructor(
        public id: string = '',
        public storageDaily: Stamp[] = [],
        public bandwidthDaily: BandwidthUsed[] = [],
        public egressDaily: EgressUsed[] = [],
        public ingressDaily: IngressUsed[] = [],
        public storageSummary: number = 0,
        public bandwidthSummary: number = 0,
        public egressSummary: number = 0,
        public ingressSummary: number = 0,
        public audit: Metric = new Metric(),
        public uptime: Metric = new Metric(),
        public joinDate: Date = new Date(),
    ) {}
}

/**
 * Stamp is storage usage stamp for satellite at some point in time
 */
export class Stamp {
    public atRestTotal: number;
    public intervalStart: Date;

    public constructor(atRestTotal: number = 0, intervalStart: Date = new Date()) {
        this.atRestTotal = atRestTotal;
        this.intervalStart = intervalStart;
    }

    /**
     * Creates new empty instance of stamp with defined date
     * @param date - holds specific date of the month
     * @returns Stamp - new empty instance of stamp with defined date
     */
    public static emptyWithDate(date: number): Stamp {
        const now = new Date();
        now.setUTCDate(date);
        now.setUTCHours(0, 0, 0, 0);

        return new Stamp(0, now);
    }
}

/**
 * Metric encapsulates storagenode reputation metrics
 */
export class Metric {
    public constructor(
        public totalCount: number = 0,
        public successCount: number = 0,
        public alpha: number = 0,
        public beta: number = 0,
        public unknownAlpha: number = 0,
        public unknownBeta: number = 0,
        public score: number = 0,
        public unknownScore: number = 0,
    ) {}
}

/**
 * Egress stores info about storage node egress usage
 */
export class Egress {
    public constructor(
        public audit: number = 0,
        public repair: number = 0,
        public usage: number = 0,
    ) {}
}

/**
 * Ingress stores info about storage node ingress usage
 */
export class Ingress {
    public constructor(
        public repair: number = 0,
        public usage: number = 0,
    ) {}
}

/**
 * BandwidthUsed stores bandwidth usage information over the period of time
 */
export class BandwidthUsed {
    public constructor(
        public egress: Egress,
        public ingress: Ingress,
        public intervalStart: Date,
    ) {}

    /**
     * Used to summarize all bandwidth usage data
     * @returns summary - sum of all bandwidth usage data
     */
    public summary(): number {
        return this.egress.audit + this.egress.repair + this.egress.usage +
        this.ingress.repair + this.ingress.usage;
    }

    /**
     * Creates new empty instance of used bandwidth with defined date
     * @param date - holds specific date of the month
     * @returns BandwidthUsed - new empty instance of used bandwidth with defined date
     */
    public static emptyWithDate(date: number): BandwidthUsed {
        const now = new Date();
        now.setUTCDate(date);
        now.setUTCHours(0, 0, 0, 0);

        return new BandwidthUsed(new Egress(0, 0, 0), new Ingress(0, 0), now);
    }
}

/**
 * EgressUsed stores egress bandwidth usage information over the period of time
 */
export class EgressUsed {
    public constructor(
        public egress: Egress,
        public intervalStart: Date,
    ) {}

    /**
     * Used to summarize all egress usage data
     * @returns summary - sum of all egress usage data
     */
    public summary(): number {
        return this.egress.audit + this.egress.repair + this.egress.usage;
    }

    /**
     * Creates new empty instance of used egress with defined date
     * @param date - holds specific date of the month
     * @returns EgressUsed - new empty instance of used egress with defined date
     */
    public static emptyWithDate(date: number): EgressUsed {
        const now = new Date();
        now.setUTCDate(date);
        now.setUTCHours(0, 0, 0, 0);

        return new EgressUsed(new Egress(0, 0, 0), now);
    }
}

/**
 * IngressUsed stores ingress usage information over the period of time
 */
export class IngressUsed {
    public constructor(
        public ingress: Ingress,
        public intervalStart: Date,
    ) {}

    /**
     * Used to summarize all ingress usage data
     * @returns summary - sum of all ingress usage data
     */
    public summary(): number {
        return this.ingress.repair + this.ingress.usage;
    }

    /**
     * Creates new empty instance of used ingress with defined date
     * @param date - holds specific date of the month
     * @returns IngressUsed - new empty instance of used ingress with defined date
     */
    public static emptyWithDate(date: number): IngressUsed {
        const now = new Date();
        now.setUTCDate(date);
        now.setUTCHours(0, 0, 0, 0);

        return new IngressUsed(new Ingress(0, 0), now);
    }
}

/**
 * Satellites encapsulate related data of all satellites
 */
export class Satellites {
    public constructor(
        public storageDaily: Stamp[] = [],
        public bandwidthDaily: BandwidthUsed[] = [],
        public egressDaily: EgressUsed[] = [],
        public ingressDaily: IngressUsed[] = [],
        public storageSummary: number = 0,
        public bandwidthSummary: number = 0,
        public egressSummary: number = 0,
        public ingressSummary: number = 0,
        public joinDate: Date = new Date(),
        public satellitesScores: SatelliteScores[] = [],
    ) {}
}

// TODO: move and create domain types.
/**
 * Holds information about audit and suspension scores by satellite.
 */
export class SatelliteScores {
    public auditScore: Score;
    public suspensionScore: Score;
    public onlineScore: Score;
    public iconClassName: string = '';

    private readonly WARNING_CLASSNAME: string = 'warning';
    private readonly DISQUALIFICATION_CLASSNAME: string = 'disqualification';

    public constructor(
        public satelliteName: string = 'satellite-name',
        auditScore: number = 0,
        unknownScore: number = 0,
        onlineScore: number = 0,
    ) {
        this.auditScore = new Score(auditScore);
        this.suspensionScore = new Score(unknownScore);
        this.onlineScore = new Score(onlineScore);
        const scores = [this.auditScore, this.onlineScore, this.suspensionScore];

        if (scores.some(score => score.statusClassName === this.DISQUALIFICATION_CLASSNAME)) {
            this.iconClassName = this.DISQUALIFICATION_CLASSNAME;

            return;
        }

        if (scores.some(score => score.statusClassName === this.WARNING_CLASSNAME)) {
            this.iconClassName = this.WARNING_CLASSNAME;
        }
    }
}

/**
 * Score in percents and className for view.
 */
export class Score {
    public label: string;
    public statusClassName: string;

    private readonly WARNING_MINIMUM_SCORE: number = 0.95;
    private readonly WARNING_CLASSNAME: string = 'warning';
    private readonly DISQUALIFICATION_MINIMUM_SCORE: number = 0.6;
    private readonly DISQUALIFICATION_CLASSNAME: string = 'disqualification';

    public constructor(
        score: number = 0,
    ) {
        this.label = `${parseFloat((score * 100).toFixed(2))} %`;

        switch (true) {
            case (score < this.DISQUALIFICATION_MINIMUM_SCORE):
                this.statusClassName = this.DISQUALIFICATION_CLASSNAME;

                break;
            case (score < this.WARNING_MINIMUM_SCORE):
                this.statusClassName = this.WARNING_CLASSNAME;

                break;
            default:
                this.statusClassName = '';
        }
    }
}
