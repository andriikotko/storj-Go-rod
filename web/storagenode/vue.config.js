// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

const path = require('path');
const StyleLintPlugin = require('stylelint-webpack-plugin');

module.exports = {
    publicPath: "/static/dist",
    productionSourceMap: false,
    parallel: true,
    configureWebpack: {
        plugins: [
            new StyleLintPlugin({
                files: ['**/*.{vue,sss,less,scss,sass}'],
                emitWarning: true,
            })
        ],
    },
    chainWebpack: config => {
        config.output.chunkFilename(`js/vendors_[hash].js`);
        config.output.filename(`js/app_[hash].js`);

        config.resolve.alias
            .set('@', path.resolve('src'));

        config
            .plugin('html')
            .tap(args => {
                args[0].template = './index.html';
                return args
            });

        const svgRule = config.module.rule('svg');

        svgRule.uses.clear();

        svgRule
            .use('babel-loader')
            .loader('babel-loader')
            .end()
            .use('vue-svg-loader')
            .loader('vue-svg-loader');
    }
};
