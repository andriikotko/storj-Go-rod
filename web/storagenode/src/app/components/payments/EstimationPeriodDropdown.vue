// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="period-container" :class="{ disabled: isCalendarDisabled }" @click.stop="openPeriodDropdown">
        <p class="period-container__label long-text">Custom Date Range</p>
        <p class="period-container__label short-text">Custom Range</p>
        <BlackArrowHide v-if="isCalendarShown" />
        <BlackArrowExpand v-else />
        <PayoutPeriodCalendar
            class="period-container__calendar"
            v-click-outside="closePeriodDropdown"
            v-if="isCalendarShown"
        />
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import PayoutPeriodCalendar from '@/app/components/payments/PayoutPeriodCalendar.vue';

import BlackArrowExpand from '@/../static/images/BlackArrowExpand.svg';
import BlackArrowHide from '@/../static/images/BlackArrowHide.svg';

import { APPSTATE_ACTIONS } from '@/app/store/modules/appState';

@Component({
    components: {
        PayoutPeriodCalendar,
        BlackArrowExpand,
        BlackArrowHide,
    },
})
export default class EstimationPeriodDropdown extends Vue {
    /**
     * Indicates if period selection calendar should appear.
     */
    public get isCalendarShown(): boolean {
        return this.$store.state.appStateModule.isPayoutCalendarShown;
    }

    /**
     * Indicates if period selection calendar should be disabled.
     */
    public get isCalendarDisabled(): boolean {
        const nodeStartedAt = this.$store.state.node.selectedSatellite.joinDate;
        const now = new Date();

        return nodeStartedAt.getUTCMonth() === now.getUTCMonth() && nodeStartedAt.getUTCFullYear() === now.getUTCFullYear();
    }

    /**
     * Opens payout period selection dropdown.
     */
    public openPeriodDropdown(): void {
        if (this.isCalendarDisabled) {
            return;
        }

        this.$store.dispatch(APPSTATE_ACTIONS.TOGGLE_PAYOUT_CALENDAR, true);
    }

    /**
     * Closes payout period selection dropdown.
     */
    public closePeriodDropdown(): void {
        this.$store.dispatch(APPSTATE_ACTIONS.TOGGLE_PAYOUT_CALENDAR, false);
    }
}
</script>

<style scoped lang="scss">
    .period-container {
        position: relative;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: center;
        background-color: transparent;
        cursor: pointer;

        &__label {
            margin-right: 8px;
            font-family: 'font_regular', sans-serif;
            font-weight: 500;
            font-size: 16px;
            color: var(--regular-text-color);
        }

        &__calendar {
            position: absolute;
            top: 30px;
            right: 0;
        }
    }

    .active {

        .period-container__label {
            color: var(--navigation-link-color);
        }
    }

    .arrow {

        path {
            fill: var(--period-selection-arrow-color);
        }
    }

    .short-text {
        display: none;
    }

    .disabled {

        .period-container {

            &__label {
                color: #909bad;
            }
        }

        .arrow {

            path {
                fill: #909bad !important;
            }
        }
    }

    @media screen and (max-width: 505px) {

        .period-container__label {
            margin-right: 4px;
        }

        .short-text {
            display: inline-block;
            font-size: 14px;
        }

        .long-text {
            display: none;
        }
    }
</style>
