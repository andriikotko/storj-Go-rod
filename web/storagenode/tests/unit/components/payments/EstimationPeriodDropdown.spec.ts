// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue from 'vue';
import Vuex from 'vuex';

import EstimationPeriodDropdown from '@/app/components/payments/EstimationPeriodDropdown.vue';

import { appStateModule } from '@/app/store/modules/appState';
import { makeNodeModule, NODE_MUTATIONS } from '@/app/store/modules/node';
import { SNOApi } from '@/storagenode/api/storagenode';
import { BandwidthInfo, Dashboard, DiskSpaceInfo, SatelliteInfo } from '@/storagenode/dashboard';
import { Metric, Satellite, Stamp } from '@/storagenode/satellite';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const nodeApi = new SNOApi();
const nodeModule = makeNodeModule(nodeApi);

const localVue = createLocalVue();
localVue.use(Vuex);

Vue.directive('click-outside', {
    bind: (): void => { return; },
    unbind: (): void => { return; },
});

const store = new Vuex.Store({ modules: { appStateModule, node: nodeModule }});

describe('DiskStatChart', (): void => {
    it('renders correctly', (): void => {
        const wrapper = shallowMount(EstimationPeriodDropdown, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('opens calendar on click only when historical data exists',   async (): Promise<void> => {
        const satelliteInfo = new Satellite(
            '3',
            [new Stamp()],
            [],
            [],
            [],
            111,
            222,
            50,
            70,
            new Metric(1, 1, 1, 0, 1, 0, 0, 1),
            new Metric(2, 1, 1, 0, 1, 0, 0, 1),
            new Date(),
        );

        const dashboardInfo = new Dashboard(
            '1',
            '2',
            [
                new SatelliteInfo('3', 'url1', null, null),
                new SatelliteInfo('4', 'url2', new Date(), new Date(2020, 0, 1)),
            ],
            new DiskSpaceInfo(99, 100, 4),
            new BandwidthInfo(50),
            new Date(),
            new Date(),
            '0.1.1',
            '0.2.2',
            false,
        );

        store.commit(NODE_MUTATIONS.POPULATE_STORE, dashboardInfo);
        store.commit(NODE_MUTATIONS.SELECT_SATELLITE, satelliteInfo);

        const wrapper = shallowMount(EstimationPeriodDropdown, {
            store,
            localVue,
        });

        wrapper.find('.period-container').trigger('click');

        await localVue.nextTick();
        expect(wrapper.find('.period-container__calendar').exists()).toBe(false);

        satelliteInfo.joinDate = new Date(Date.UTC(2019, 10, 29));

        store.commit(NODE_MUTATIONS.SELECT_SATELLITE, satelliteInfo);

        wrapper.find('.period-container').trigger('click');

        await localVue.nextTick();
        expect(wrapper.find('.period-container__calendar').exists()).toBe(true);
    });
});
