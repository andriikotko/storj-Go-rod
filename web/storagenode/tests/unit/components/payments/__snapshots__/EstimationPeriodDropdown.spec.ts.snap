// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiskStatChart renders correctly 1`] = `
<div class="period-container disabled">
  <p class="period-container__label long-text">Custom Date Range</p>
  <p class="period-container__label short-text">Custom Range</p>
  <blackarrowexpand-stub></blackarrowexpand-stub>
  <!---->
</div>
`;
